kaolin/helpers.py

```diff
@@ -177,50 +177,54 @@ def _get_hash(x):
 
 
 class Cache(object):
-    """Caches the results of the called function to disk.
+    """Caches the results of a function to disk.
     If already cached, data is returned from disk, otherwise,
-    the function called is executed.
+    the function is executed. Output tensors are always on CPU device.
 
         Args:
             transforms (Iterable): List of transforms to compose.
             cache_dir (str): Directory where objects will be cached. Default
                              to 'cache'.
     """
 
-    def __init__(self, func: Callable, cache_dir: str = 'cache', cache_key: str = ''):
+    def __init__(self, func: Callable, cache_dir: [str, Path], cache_key: str):
+        assert isinstance(cache_dir, str) or isinstance(cache_dir, Path)
+        assert isinstance(cache_key, str)
+        
+        # Assert cache dir and key are not empty
+        assert str(cache_dir).strip()
+        assert cache_key.strip()
```

Comment:
Agree path directory could be empty. However, the key should not since an empty key would likely only occur if the user forgot to pass it in and it would result in every item referring to the same data. I'd therefore leave it in just as a check, if you agree.

Reply:
`cache_key` have no default value, so the user have to pass a value (which is good) otherwise calling the function would trigger a `TypeError: __init__() is missing 1 required positional argument ...`

If you want to exclude the `''` I think we should also exclude `'.'`, which would pass the current assertion, but would have the exact same effect.

Reply:
Valid. I'll remove that check too then since it doesn't in fact improve usability. Thanks!

Reply:
I removed the check to for a str type as well and cast to str when setting the `self.cache_dir` value. This will allow objects with a `__repr__` that can be used as a key to be passed in without error.